File: api/FilterLists.Api/Api/Lists/GetSummaries/Response/FilterListSummary.cs
using System.Collections.Generic;

namespace FilterLists.Api.Api.Lists.GetSummaries.Response;

public record FilterListSummary
{
    public string Name { get; init; } = default!;
    public string? Description { get; init; }
    public IEnumerable<string> Software { get; init; } = new HashSet<string>();
    public IEnumerable<string> Syntaxes { get; init; } = new HashSet<string>();
    public IEnumerable<string> Languages { get; init; } = new HashSet<string>();
    public IEnumerable<string> Tags { get; init; } = new HashSet<string>();
    public IEnumerable<string> Maintainers { get; init; } = new HashSet<string>();
    public string License { get; init; } = default!;
}

File: web/apps/web/src/app/services/filter-list-summary.ts
export interface FilterListSummary {
  id: number;
  name: string;
  description: string;
  languages: Language[];
  license: string;
  maintainers: string[];
  software: Software[];
  syntaxes: Syntax[];
  tags: Tag[];
}

export interface Language {
  iso6391: string;
  name: string;
}

export interface Software {
  id: number;
  name: string;
}

export interface Syntax {
  name: string;
  description: string;
}

export interface Tag {
  name: string;
  description: string;
}
